common/sc_popen.cpp

```diff
@@ -163,10 +163,11 @@ std::tuple<pid_t, FILE*> sc_popen_argv(const std::vector<std::string>& strings,
         strings.begin(), strings.end(), std::string(),
         [](const std::string& a, const std::string& b) -> std::string { return a + (a.length() > 0 ? " " : "") + b; });
 
-    return sc_popen(std::move(commandLine), type);
+    return sc_popen_c(commandLine.data(), type.data());
 }
 
-std::tuple<pid_t, FILE*> sc_popen(std::string&& command, const std::string& type) {
+std::tuple<pid_t, FILE*> sc_popen_shell(std::string&& command, const std::string& type) {
+    command = "cmd /c " + command;
```

Comment:
Modifying and reassinging to an rvalue reference is a bit funky. I think `sc_popen_shell` should just take `std::string` by value.

Generally, functions should only take arguments by rvalue reference if it *must* be an r-value reference, e.g. to distinguish a copy constructor from a move constructor. Another use case are "unversial references":
```
template<typename T>
void foo(T&& x);
```
Otherwise, just pass by value. The caller can still move into the function and the function itself can move the argument.

Reply:
I know this is basic stuff, but it goes beyond my knowledge of cpp. I tried applying the change, is this what you meant?

Reply:
Also, I noticed that `sc_popen_shell()` on *nix uses moves the command string when passing the data to the vector. Is that done "just in case" or is there an actual reason for it?

Reply:
> I tried applying the change, is this what you meant?

Yes, thanks!

> Also, I noticed that sc_popen_shell() on *nix moves the command string when passing the data to the vector. Is that done "just in case" or is there an actual reason for it?

This is just a small performance optimization. `std::move` turns a normal reference into a rvalue-reference, so that the destination - here `std::vector::push_back` - knows that it can move from its argument. In case of `std::string` it can steal the underlying buffer instead of making a copy. (Does this actually matter in the context of spawning a subprocess? Probably not :)